ToolChain/Assembler/Assembler.py

```diff
@@ -56,79 +50,90 @@ def __init__(self, inputFile=None, outputFile=None):
-                if wouldBeExported in self.parser.referenceDict.keys():
-                    self.parser.referenceDict[wouldBeExported] = self.parser.referenceDict[reference]
-
-        fileContent += b"</InternalSymbols>"
-
-        # External symbols allow for a file to be linked with another file
-        fileContent += b"<ExternalSymbols>"
-        for reference in self.parser.referenceDict:
-            if EXPORTED_REFERENCE_INDICATOR in reference:
-                isValid = False
-                shortReference = reference.split()[1]
-                for internalReference in validInternalRef:
-                    if shortReference == internalReference:
-                        isValid = True
-                        break
-
-                if isValid:
-                    refName = reference.split()[1]
-                    fileContent += b"<refName>" + refName.encode("utf-8") + b"</refName>"
-                    fileContent += b"<refAdd>" + struct.pack(">I", self.parser.referenceDict[reference]) + b"</refAdd>"
-                else:
-                    raise ValueError("Unresolved reference: {} is exported but "
-                                     "is never declared.".format(reference))
-        fileContent += b"</ExternalSymbols>"
-
-        # Deal with the text section
-        fileContent += b"<Text>"
-        for instruction in self.parser.instructionList:
-            fileContent += instruction
-        fileContent += b"</Text>"
-
-        return fileContent
-
-    def writeAssembledFile(self, file=None, outputData=None):
-        """
-        This will write the assembled file to disk
-        :param file: str, output file path
-        :param outputData: bytes, the data to be written on the disk
+        This method assembles the binary data itself. It will parse each line individually and extract the info needed:
+        1.If the label flag is set to 0, it's a regular instruction, string, number, of memref.
+        2.If the label flag is set to 1, it's an internal symbol.
+        3.If the label flag is set to 2, it's an external symbol.
+        In cases 2 and 3, the symbols are added to their respective lists along with their memory location.
+        Memory locations are relative to the beginning of the file. Instructions and data are given set lengths.
+        Internal and external symbol lists contain tuples containing the symbol name, and its offset.
+        These will be used to fill the .o file information before the binary data.
+        :param inputFile: str, the name of the file to be read.
+        :param outputFile: str, the name of the file to be written to.
         :return:
         """
 
-        file = open(file, mode="wb")
-        file.write(outputData)
+        parser = Parser()                    # Parser object which will return the binary code for instructions
+        masterString = b""                   # String that will eventually be our output file
```

Comment:
Comments here are misleading. The xmlstring is the variable that ends up being the full file. masterString only contains the text section of the file.